torch/_subclasses/fake_tensor.py

```diff
@@ -657,24 +690,29 @@ def to_real_tensor(e):
 
         for e in tree_flatten((args, kwargs))[0]:
             if isinstance(e, torch.Tensor):
-                storages.add(e.storage()._cdata)
+                if not e.is_sparse:
+                    storages.add(e.storage()._cdata)
 
         # TODO: also check metadata change on inputs
         # proper aliasing/metadata relationship between outputs and inputs will
         # not be set up, bc of conversion to device, unless we can reuse an
         # input impl
         for e in tree_flatten(r)[0]:
             if id(e) not in inp_impls and (
-                isinstance(e, torch.Tensor) and e.storage()._cdata in storages
+                isinstance(e, torch.Tensor)
+                and not e.is_sparse
+                and e.storage()._cdata in storages
             ):
                 raise orig_not_implemented_exception
 
-    # the outputs which are are not reused from impls will be converted
-    # to fake tensors later
-    meta_converter = MetaConverter()
-
     def map_out(e):
-        return inp_impls.get(id(e), meta_converter(e))
+        if isinstance(e, torch.Tensor):
+            if id(e) in inp_impls:
+                return inp_impls[id(e)]
+            else:
+                return fake_mode.fake_tensor_converter(fake_mode, e)
```

Comment:
Hmm, why is this change necessary ? I guess this works now because we're using the actual devices, not cpu fallback.

Reply:
So, the problem I was having is that the meta converter would sometimes fail to convert the output, resulting in a non meta tensor, which would then fail later. It seemed cleaner to go straight to fake tensor here.